core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -4476,6 +4476,85 @@ public String getDefaultScheme() {
     }
   }
 
+  @Test
+  public void testApplyDefaultServiceConfigIfInitialNameResolutionFails() throws Exception {
+    LoadBalancerRegistry.getDefaultRegistry().register(mockLoadBalancerProvider);
+    timer.forwardNanos(1234);
+    channelBuilder.maxTraceEvents(10);
+    try {
+      FakeNameResolverFactory nameResolverFactory = new FakeNameResolverFactory
+              .Builder(expectedUri).setServers(Collections.singletonList(new
+              EquivalentAddressGroup(socketAddress))).build();
+      channelBuilder.nameResolverFactory(nameResolverFactory);
+      Map<String, Object> defaultServiceConfig =
+              parseConfig("{\"methodConfig\":[{"
+                      + "\"name\":[{\"service\":\"SimpleService1\"}],"
+                      + "\"waitForReady\":true}]}");
+      channelBuilder.defaultServiceConfig(defaultServiceConfig);
+      createChannel();
+      FakeNameResolverFactory.FakeNameResolver resolver = nameResolverFactory.resolvers.get(0);
+      Status resolutionError = Status.UNAVAILABLE
+              .withDescription("Initial Name Resolution error, using default service config");
+
+      int prevSize = getStats(channel).channelTrace.events.size();
+      resolver.listener.onError(resolutionError);
+      verify(mockLoadBalancer).handleNameResolutionError(resolutionError);
+
+      assertThat(getStats(channel).channelTrace.events).hasSize(prevSize + 2);
+      assertThat(getStats(channel).channelTrace.events).contains(new ChannelTrace.Event.Builder()
```

Comment:
Instead of contains, use the index prevSize - 2 to retrieve this trace event since that is more specific. Also add the assertion for the last element (at prevSize - 1) that it contains the trace event for applying defalut service config.

Reply:
Fixed